File: update-core/src/test/java/dev/bee/kanjianki/updatecore/UpdateRunScreenCopyTest.java
package dev.bee.kanjianki.updatecore;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public final class UpdateRunScreenCopyTest {
    @Test
    public void cachedPendingRunPreservesInstallerCopy() {
        UpdateRunScreenCopy.Copy copy = UpdateRunScreenCopy.forRun(true);

        assertEquals("Starting installer", copy.title());
        assertEquals("Using the verified APK already cached by Kani.", copy.body());
        assertEquals("Preparing verified APK", copy.progressLabel());
    }

    @Test
    public void manualRunPreservesReleaseCheckCopy() {
        UpdateRunScreenCopy.Copy copy = UpdateRunScreenCopy.forRun(false);

        assertEquals("Checking release", copy.title());
        assertEquals("Downloading metadata and verifying assets.", copy.body());
        assertEquals("Checking GitHub Releases", copy.progressLabel());
    }
}

File: update-core/src/test/kotlin/dev/bee/kanjianki/updatecore/UpdateRunScreenCopyTest.kt
package dev.bee.kanjianki.updatecore

import org.junit.Assert.assertEquals
import org.junit.Test
import java.util.Locale

class UpdateRunScreenCopyTest {
    @Test
    fun cachedPendingRunPreservesInstallerCopy() {
        val copy = UpdateRunScreenCopy.forRun(true)

        assertEquals("Preparing installer", copy.title())
        assertEquals("Verifying APK", copy.progressLabel())
    }

    @Test
    fun manualRunPreservesReleaseCheckCopy() {
        val copy = UpdateRunScreenCopy.forRun(false)

        assertEquals("Checking for updates", copy.title())
        assertEquals("Checking releases", copy.progressLabel())
    }

    @Test
    fun japaneseLocaleTranslatesUpdateRunCopy() {
        withLocale(Locale.JAPANESE) {
            val cachedCopy = UpdateRunScreenCopy.forRun(true)
            val manualCopy = UpdateRunScreenCopy.forRun(false)

            assertEquals("インストーラーを準備中", cachedCopy.title())
            assertEquals("APKを確認中", cachedCopy.progressLabel())
            assertEquals("更新を確認中", manualCopy.title())
            assertEquals("リリースを確認中", manualCopy.progressLabel())
        }
    }

    private inline fun <T> withLocale(locale: Locale, block: () -> T): T {
        val previous = Locale.getDefault()
        Locale.setDefault(locale)
        return try {
            block()
        } finally {
            Locale.setDefault(previous)
        }
    }
}
